src/data/entities/D2ExpressionParser.ts

```diff
@@ -0,0 +1,114 @@
+import * as xp from "@dhis2/expression-parser";
+import _c from "../../domain/entities/generic/Collection";
+import { Either } from "../../domain/entities/generic/Either";
+
+export type ProgramRuleVariableType = "text" | "number" | "date" | "boolean";
+
+export type ProgramRuleVariableName = string;
+export type ProgramRuleVariableValue = {
+    type: ProgramRuleVariableType;
+    value: string;
+};
+
+const VariableValueTypeMap: Record<ProgramRuleVariableType, xp.ValueType> = {
+    text: xp.ValueType.STRING,
+    boolean: xp.ValueType.BOOLEAN,
+    date: xp.ValueType.DATE,
+    number: xp.ValueType.NUMBER,
+};
```

Comment:
Let's move auxiliar/helper methods down, so the first we see when opening the file is the class and its public methods.